pxteditor/editor.ts

```diff
@@ -257,6 +257,7 @@ namespace pxt.editor {
         showUploadInstructionsAsync?: (fn: string, url: string, confirmAsync: (options: any) => Promise<number>) => Promise<void>;
         fieldEditors?: IFieldCustomOptions[];
         toolboxOptions?: IToolboxOptions;
+        winrtHandleDeviceNotFoundAsync?: (io: pxt.HF2.PacketIO) => Promise<void>
```

Comment:
can we make this work for any HID based onnection? (not just winrt)?

Reply:
Sure but the other scenarios are not user facing and are in independent code paths, so they can be done in a separate change

Reply:
Why is this WinRT specific? Couldn't we could also face this via webusb as well? Just suggesting to make this callback a bit more generic so that it can be reused in other platforms.

Reply:
I don't think we can make it too generic, everything inside that callback on the Adafruit needs to use WinRT-specific APIs:
- Reconnect via WinRT serial
- Set the baudrate using WinRT SerialDevice API
- WinRT-specific device disconnect
- Re-init the WindowsRuntimePacketIo (`initAsync` is not standard in our base PacketIo class)

I'm open to suggestions if you can think of how to design a single generic WinRT / WebUSB extension point. Otherwise I suggest we use 2 separate callbacks, and then the target can refactor some common logic between the 2 callbacks if possible / necessary